src/log_batch.rs

```diff
@@ -779,14 +790,17 @@ impl LogBatch {
         Ok(self.buf.len() - header_offset)
     }
 
-    /// Make preparations for the write of `LogBatch`.
+    /// Makes preparations for the write of `LogBatch`.
     #[inline]
     pub(crate) fn prepare_write(&mut self, file_context: &LogFileContext) -> Result<()> {
         match self.buf_state {
             BufState::Encoded(header_offset, entries_len) => {
                 LogItemBatch::prepare_write(&mut self.buf, file_context)?;
                 self.buf_state = BufState::Sealed(header_offset, entries_len);
             }
+            BufState::ReSealing(_, _) => {
+                LogItemBatch::prepare_write(&mut self.buf, file_context)?;
```

Comment:
This doesn't make sense, the signature will be `checksum^first_file^second_file`. I wonder why the tests didn't catch it.